sdk/cosmos/src/clients/cosmos_struct.rs

```diff
@@ -5,200 +5,108 @@ use crate::{
-            auth_token,
-            cosmos_uri_builder: self.cosmos_uri_builder.clone(),
-        }
+impl CosmosStruct {
+    /// Create a new `CosmosStruct` which connects to the account's instance in the public Azure cloud.
+    pub fn new(account: String, auth_token: AuthorizationToken) -> CosmosStruct {
+        let cloud_location = CloudLocation::Public(account);
+        Self::new_with_cloud_location(cloud_location, auth_token)
     }
-}
 
-#[derive(Debug, Clone)]
-pub struct DefaultCosmosUri {
-    uri: String,
-}
-
-impl DefaultCosmosUri {
-    fn new(account: &str) -> DefaultCosmosUri {
-        DefaultCosmosUri {
-            uri: format!("https://{}.documents.azure.com", account),
-        }
+    pub fn new_china(account: String, auth_token: AuthorizationToken) -> Self {
+        let cloud_location = CloudLocation::China(account);
+        Self::new_with_cloud_location(cloud_location, auth_token)
     }
-}
-
-impl CosmosUriBuilder for DefaultCosmosUri {
-    fn build_base_uri(&self) -> &str {
-        &self.uri
-    }
-}
-
-#[derive(Debug, Clone, Default)]
-pub struct ChinaCosmosUri {
-    uri: String,
-}
-
-impl ChinaCosmosUri {
-    fn new(account: &str) -> ChinaCosmosUri {
-        ChinaCosmosUri {
-            uri: format!("https://{}.documents.azure.cn", account),
-        }
-    }
-}
-
-impl CosmosUriBuilder for ChinaCosmosUri {
-    fn build_base_uri(&self) -> &str {
-        &self.uri
-    }
-}
 
-#[derive(Debug, Clone, Default)]
-pub struct CustomCosmosUri {
-    uri: String,
-}
-
-impl CosmosUriBuilder for CustomCosmosUri {
-    #[inline]
-    fn build_base_uri(&self) -> &str {
-        &self.uri
+    pub fn new_custom(account: String, auth_token: AuthorizationToken, uri: String) -> Self {
+        let cloud_location = CloudLocation::Custom { account, uri };
+        Self::new_with_cloud_location(cloud_location, auth_token)
     }
-}
-
-pub struct ClientBuilder {}
-
-impl ClientBuilder {
-    #[allow(clippy::new_ret_no_self)]
-    pub fn new<'a, IntoCowStr>(
-        account: IntoCowStr,
-        auth_token: AuthorizationToken,
-    ) -> Result<CosmosStruct<'a, DefaultCosmosUri>, AzureError>
-    where
-        IntoCowStr: Into<Cow<'a, str>>,
-    {
-        let account = account.into();
-        let client = hyper::Client::builder().build(HttpsConnector::new());
-        let cosmos_uri_builder = DefaultCosmosUri::new(account.as_ref());
 
-        Ok(CosmosStruct {
-            hyper_client: client,
-            account: account,
-            auth_token,
-            cosmos_uri_builder,
-        })
+    pub fn new_emulator(address: &str, port: u16) -> Self {
+        //Account name: localhost:<port>
+        //Account key: C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==
+        let auth_token = AuthorizationToken::new_master(
+            "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==",
+        ).unwrap();
```

Comment:
Remove?

Reply:
Remove the unwrap? I don't believe we can. Can you explain?

Reply:
Never mind, just realized that this was a magic string for the Cosmos DB emulator account key, I thought this was a leftover from debugging. Ignore my silliness 🙃

Small suggestion, feel free to ignore if out of scope for the PR - document and refer to [the docs](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator) for the emulator. Maybe move that connection string to a const. Might avoid some confusion for the uninitiated like myself.